DEVELOPMENT.md

```diff
@@ -179,3 +179,43 @@ Here is a short summary of the various commands and their primary function. Plea
 
     However, you cannot use `ibazel run` in this case. The file watcher is glitchy on running the tests
     when detecting changes. It shows '`a broken pipe`' in terminal. We need to terminate and restart the program manually.
+
+## Adding, Updating, or Removing Frontend Dependencies
+
+For the most part, frontend-specific third-party dependencies are hosted by
+[npm](https://docs.npmjs.com/about-npm) and managed by
+[yarn](https://classic.yarnpkg.com/). Some of our build-time dependencies are
+managed by [bazel](https://bazel.build/).
+
+The source of truth for `yarn` is a combination of the `package.json` and
+`yarn.lock` files. `package.json` is maintained by us developers and describes
+direct dependencies while `yarn.lock` is maintained by the system and describes
+the entire tree of transitive dependencies. In reality both files are often
+edited automatically by calls to `yarn` and `package.json` only sometimes needs
+to be edited by hand.
+
+1.  Install yarn.
```

Comment:
Maybe a reference would be useful https://classic.yarnpkg.com/lang/en/docs/install/#mac-stable

Reply:
Done.